### DISCUSSION TITLE
Integrating SSL into SegmentationTask

### DISCUSSION THREAD

**Discussion by Dodekaphonia** [torchgeo/torchgeo#1937] [Q&A]
Hello!

I'm currently working on a model for coastline change detection along the Antarctic coastline using Sentinel-1 images.

I followed @calebrob6's tutorial on [LEVIR+](https://gist.github.com/calebrob6/4d7fc311045c4f9015e401100e34ed38) and tested this approach on my relatively small dataset of labeled changes, utilizing the `ResNet50_Weights.SENTINEL1_ALL_MOCO` weights. The results appeared promising.

Considering further improvements, I considered the idea of pre-training a self-supervised learning (SSL) model and then using it as a backbone for the `SegmentationTask`. To stay within one framework, I attempted to utilize the [BYOLTask](https://torchgeo.readthedocs.io/en/stable/api/trainers.html#torchgeo.trainers.BYOLTask), [MoCoTask](https://torchgeo.readthedocs.io/en/stable/api/trainers.html#torchgeo.trainers.MoCoTask), and [SimCLRTask](https://torchgeo.readthedocs.io/en/stable/api/trainers.html#torchgeo.trainers.SimCLRTask). However, I encountered difficulties in finding a straightforward approach to achieve pass the '.ckpt' file into a Segmentation

So, my question is whether it is possible to pre-train one of the SSL tasks and then integrate it into the Segmentation process. Or should I explore alternative approaches?

**Comment by adamjstewart** [comment]
I feel like @isaaccorley would know a good way to do this.

**Reply by isaaccorley** [reply]
I'm working on a solution for this. Stay tuned. We have some checkpoint loading utils but I haven't used them to load a BYOL checkpoint in a few months.
